### ISSUE TITLE
Inconsistent timezone handling for time values

### ISSUE THREAD

**Issue by paulrosenzweig** [metabase/metabase#10366]
## TL;DR

Timezones for time values are handled differently from other date/time values. This leads to errors in how we display times.

## Example Setup

Here's the schema for a postgres table I created to test this. Each column is named for its type.
```
   Column    |            Type
-------------+-----------------------------
 time        | time without time zone
 timetz      | time with time zone
 timestamp   | timestamp without time zone
 timestamptz | timestamp with time zone
 date        | date
```

I inserted `'2000-01-01 00:00:00'` into each column. This is the resulting row printed in psql:
```
   time   |   timetz    |      timestamp      |      timestamptz       |    date    
----------+-------------+---------------------+------------------------+------------
 00:00:00 | 00:00:00-05 | 2000-01-01 00:00:00 | 2000-01-01 00:00:00-05 | 2000-01-01
```

(The db is in US/Eastern)

## Server Responses

I queried this table with different values for the "Report Timezone" option. These are the values returned from the API.

**Report Timezone: Database Default**
```
date        "2000-01-01T00:00:00.000-05:00"
time                   "19:00:00.000-05:00" 👈19:00 rather than 00:00
timestamp   "2000-01-01T00:00:00.000-05:00"
timestamptz "2000-01-01T00:00:00.000-05:00"
timetz                 "00:00:00.000-05:00"
```

**Report Timezone: UTC**
```
date        "2000-01-01T00:00:00.000Z"
time                   "00:00:00.000Z"
timestamp   "2000-01-01T00:00:00.000Z"
timestamptz "2000-01-01T05:00:00.000Z"
timetz                 "05:00:00.000Z"
```

**Report Timezone: US/Eastern**
```
date        "2000-01-01T00:00:00.000-05:00"
time                   "19:00:00.000-05:00" 👈19:00 rather than 00:00
timestamp   "2000-01-01T00:00:00.000-05:00"
timestamptz "2000-01-01T00:00:00.000-05:00"
timetz                 "00:00:00.000-05:00"
```

## The Problem

With the report timezone set to UTC everything works. Set to the local timezone or using the default, the "time" column is inconsistent with all the others.

Adjusting the reporting timezone adjusts the returned timezone for `date`, `timestamp`, `timestamptz`, and `timetz`. The date/time returned is not changed - only the timezone. For `time` values however, both the timezone and the time value itself are changed.

This leads "7pm" showing up in the UI for `time` values but "12am" showing up for all other types.

The example is done in postgres. The exact types are different in mysql, but the behavior for `time` values is the same.

**Comment by florianmonfort** [comment]
Hope you don't mind if I subscribe to follow that particular issue 😛

**Comment by camsaul** [comment]
need to double-check this has a test

**Comment by camsaul** [comment]
Fixed by #11280
